packages/block-editor/src/components/media-upload-progress/index.native.js

```diff
@@ -115,19 +117,26 @@ export class MediaUploadProgress extends Component {
 		}
 	}
 
+	getRetryMessage() {
+		if ( this.props.isConnected === true ) {
+			// eslint-disable-next-line @wordpress/i18n-no-collapsible-whitespace
+			return __( 'Failed to insert media.\nTap for more info.' );
+		}
+		return __( 'You are currently offline.' );
+	}
+
 	render() {
 		const { renderContent = () => null } = this.props;
 		const { isUploadInProgress, isUploadFailed } = this.state;
 		const showSpinner = this.state.isUploadInProgress;
 		const progress = this.state.progress * 100;
-		// eslint-disable-next-line @wordpress/i18n-no-collapsible-whitespace
-		const retryMessage = __(
-			'Failed to insert media.\nTap for more info.'
-		);
+		const retryMessage = this.getRetryMessage();
 
 		const progressBarStyle = [
 			styles.progressBar,
-			showSpinner || styles.progressBarHidden,
+			showSpinner && this.props.isConnected
+				? styles.progressBarHidden
+				: null,
```

Comment:
In the current `trunk`, the entire progress bar becomes transparent via `styles.progressBarHidden`. When offline, this logic will persist the progressBar _track_, but as far as I can tell, the actual `progress` value is reset to `0` when `MEDIA_UPLOAD_STATE_FAILED` is thrown.

I have not yet uncovered where this progress reset occurs, but I believe it may be either in the host apps code or in the react-native-bridge glue code. Ideally we would not reset the progress value to `0` when `MEDIA_UPLOAD_STATE_FAILED` is thrown, but persist it and assume that the upload may be resumed. I also entertained the idea of introducing a new state like `MEDIA_UPLOAD_STATE_PAUSED` for when the app is offline, but that may be out of scope here.